components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/util/IdentityContextUtil.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.context.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.base.IdentityConstants;
+import org.wso2.carbon.identity.core.context.IdentityContext;
+import org.wso2.carbon.identity.core.context.model.Header;
+import org.wso2.carbon.identity.core.context.model.Request;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for IdentityContext related operations.
+ */
+public class IdentityContextUtil {
+
+    private static final Log LOG = LogFactory.getLog(IdentityContextUtil.class);
+
+    /**
+     * Resolves the client IP address from the given IdentityContext.
+     *
+     * @param identityContext The IdentityContext object.
+     * @return The client IP address if found, else falls back to the request IP address.
+     */
+    public static String getClientIpAddress(IdentityContext identityContext) {
```

Comment:
Do we need to pass the identity context to this method ?
Can't we directly access the identity context inside the method ?